crates/cast/src/args.rs

```diff
@@ -195,8 +195,15 @@ pub async fn run_command(args: CastArgs) -> Result<()> {
             let tokens = SimpleCast::calldata_decode(&sig, &calldata, true)?;
             print_tokens(&tokens);
         }
-        CastSubcommand::CalldataEncode { sig, args } => {
-            sh_println!("{}", SimpleCast::calldata_encode(sig, &args)?)?;
+        CastSubcommand::CalldataEncode { sig, args, file } => {
+            let final_args = if let Some(file_path) = file {
+                let contents = fs::read_to_string(file_path)?;
+                let contents = contents.trim();
+                vec![contents.to_string()]
```

Comment:
This is not correct as it treats the entire file as a single argument. It should split the string into arguments using `.lines().collect()`. This should also be done above in AbiEncode since it's a similar command

Reply:
cool , done